registry-core/src/main/java/com/tesco/aqueduct/registry/PostgreSQLTillStorage.java

```
@@ -0,0 +1,71 @@
+package com.tesco.aqueduct.registry;
+
+import com.tesco.aqueduct.registry.model.Till;
+import org.slf4j.LoggerFactory;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.ZoneOffset;
+
+public class PostgreSQLTillStorage implements TillStorage {
+
+    private final DataSource dataSource;
+    private static final RegistryLogger LOG = new RegistryLogger(LoggerFactory.getLogger(PostgreSQLTillStorage.class));
+    private static final String QUERY_INSERT_TILL =
+        "INSERT INTO tills (host_id, bootstrap_requested, bootstrap_type)" +
+            "VALUES (" +
+            "?, " +
+            "?, " +
+            "? " +
+            ")" +
+        "ON CONFLICT (host_id) DO UPDATE SET " +
+            "host_id = EXCLUDED.host_id, " +
+            "bootstrap_requested = EXCLUDED.bootstrap_requested, " +
+            "bootstrap_type = EXCLUDED.bootstrap_type, " +
+            "bootstrap_received = null;";
+
+    public PostgreSQLTillStorage(final DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    @Override
+    public void updateTill(Till till) {
+         try (Connection connection = getConnection()) {
+             insert(connection, till);
+         } catch (SQLException exception) {
+             LOG.error("Postgresql till storage", "hostId", exception);
```

Hiding the error - the caller will think the bootstrap request was successful when it wasn't. Should be throwing this error up the chain, and turning it into a 500 status.

---

👍 good catch